lib/rubygems/version.rb

```diff
@@ -279,7 +279,10 @@ def encode_with coder # :nodoc:
   # A version is considered a prerelease if it contains a letter.
 
   def prerelease?
-    @prerelease ||= !!(@version =~ /[a-zA-Z]/)
+    unless defined?(@prerelease)
```

Comment:
Can you use `instance_variable_defined?` instead? I prefer the most-specific check possible.